examples/data-hooks/rank-batch-records-compute.ts

```diff
@@ -0,0 +1,140 @@
+ * 
+ * Notice the definitions below that look like:
+ * 
+ * type RecordInput = { 
+ *   id: number,
+ *   revenue: number,
+ *   geoCode: string,
+ *   industry: string
+ * }
+ *
+ * type RecordsInputs = Array<RecordInput>
+ * 
+ * ...and similar ones for: RecordOutput and RecordsOutput
+ * 
+ * These are aliases that will be used in the main export function to help with readability
+ * 
+ * Like this:
+ * 
+ * function RankBatchRecordCompute(records: RecordsInputs): RecordsOutputs {
+ *   return calculateRecordsWithRank(records)
+ * }
+ * 
+ * without Type Aliases, we would define input and output types in the function... 
+ * which is a bit harder to read:
+ * 
+ * * function RankBatchRecordCompute(
+ *     records: Array<object: { id: number, revenue: number, geoCode: string, industry: string }
+ *   ): Array<object: { id: number, rank: number } {
+ *   return calculateRecordsWithRank(records)
+ * }
+ * 
+ * Read along through this example and notice how Typescript is just Javascript with some type checking.
+ * ===================================
+ */
+
+ 
+/**
+ * @type {Object} - an input object with keys id, revenue, geoCode and industry
+ */
+type RecordInput = { 
+  id: number,
+  revenue: number,
+  geoCode: string,
+  industry: string
+ }
+
+/**
+ * @type {Array<RecordInput>} - an array of objects that match the type alias RecordInput
+ */
+type RecordsInputs = Array<RecordInput>
+
+/**
+ * @type {Object} - an output object with keys id and rank
+ */
+type RecordOutput = { 
+  id: number,
+  rank: number
+ }
+
+/**
+ * @type {Array<RecordOutput>} - an array of objects that match the type alias RecordOutput
+ */
+type RecordsOutputs = Array<RecordOutput>
+
+
+/**
+ * @param {RecordsInput} records - input records array of objects that uses id, revenue, geoCode and industry
+ * @return {RecordOutput} returns array of record objects with id, rank
+ */
+ export default function RankBatchRecordCompute(records: RecordsInputs): RecordsOutputs {
+  return calculateRecordsWithRank(records)
+ }
+
+/**
+ * functions represent fictional business logic to determine a ranked score
+ * comparison against all records from an import
+ */
+function calculateRecordsWithRank(records: RecordsInputs): RecordsOutputs {
+  // 1. Set initial score to each record based on business rules
+  const recordsWithScore = records.map(record => {
+    const score = getScore(record.revenue, record.geoCode, record.industry)
+    return {id: record.id, score }
+  })
+
+  // 2. Sort array based on comparison of baseScores 
+  const sortedRecords = recordsWithScore.sort(function(a, b){ 
+    return a.score - b.score
+  })
+
+  // 3. return record id and 'rank' based on sort index. Ommit temp score
+  const recordsWithRank = sortedRecords.map((record, index) => {
+    const rank = index + 1
+    return {id: record.id, rank}
+  })
+
+  return recordsWithRank
+}
```

Comment:
Ranking records inside a batch is confusing for users because a 1000 record batch isn't necessarily the entire sheet.

Reply:
Sounds good... do you think it makes sense to continue building out an example like this to help illustrate when/how to use batch?  (..and would make it properly handle all the records, not just 1,000).

Reply:
@paddymul my understanding is that this is an improper use of batch compute because it's trying to do something that requires the entire set of records (ranking them) when we only have access to batches of 1000. Correct me if I'm wrong.

The example you always use for batch compute is an HTTP request. Should we use that here instead of ranking? Or is there another use case you can think of for batch compute?

Reply:
I'm going to remove this entirely for now in favor of getting everything else in. I feel I still don't fully understand the use cases for batch compute.

Reply:
BatchCompute should only be used for making async calls, (http calls).  BatchCompute only operates on 1000 row chunks, not necessarily the entire sheet.  To operate on the entire sheet you need to use `sheetCompute`.  most sorting in a data hook is some type of adhoc grouping anyway